### PR TITLE
[Feature] Switch to localforage

### PR DESCRIPTION
**Pull request by Mogball**
I added my fork as a package `ember-localforage`.

The owner of `ember-local-forage` said he'd review my PR and if he accepts the changes and merges it in we can just change back to `ember-local-forage`.

### CODE REVIEW

File: app/services/query-manager.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ export default Ember.Service.extend({
 
   addResult(id, data) {
     return this.get('store').findRecord('query', id).then((query) => {
-      let result = this.get('store').createRecord('result', {
+      let result = Ember.run(this.get('store'), 'createRecord', 'result', {
```
Comment (akshaisarma) [inline app/services/query-manager.js]: Why do we need another run loop again?

Comment (Mogball) [reply app/services/query-manager.js]: I forget; let me see if it works without

**Comment by Mogball** [comment]
@akshaisarma Tests use localstorage because although the adapter is requesting indexeddb, for whatever reason phantomjs isn't able to provide it so `ember-localforage` will fall back to localStorage

**Comment by akshaisarma** [comment]
Thanks! Would you mind fixing [migrations.js](/yahoo/bullet-ui/blob/master/app/instance-initializers/migrations.js) and test to account for localforage?

**Comment by Mogball** [comment]
apparently it's only available in a beta version not on npm yet :rip:

File: tests/unit/instance-initializers/migrations-test.js [hunk 2]
```diff
@@ -51,9 +51,12 @@ test('it applies the delete results migration', function(assert) {
 test('it applies the delete queries migration', function(assert) {
   assert.expect(1);
 
+  let manager = {
+    deleteAllQueries() {
+      assert.ok(true);
+    }
+  }
```
Comment (akshaisarma) [inline tests/unit/instance-initializers/migrations-test.js]: Semicolon. JSHint should catch this.

**Comment by akshaisarma** [comment]
Thanks, @Mogball! 🥇

**Pull request merged**